vendor/engines/split_accounts/app/services/split_accounts/split_order_detail_simulator.rb

```diff
@@ -0,0 +1,90 @@
+module SplitAccounts
+
+  # TODO: maybe refactor some of this into SplitSimulator
+  # TODO: we can refactor this code to fewer lines and iterate fewer times
+  class SplitOrderDetailSimulator
```

Comment:
I'm not sure I love the term `Simulator`. I don't really have anything better, though.